subsys/pm/pm.c

```diff
@@ -194,7 +194,16 @@ bool pm_system_suspend(int32_t ticks)
 	SYS_PORT_TRACING_FUNC_ENTER(pm, system_suspend, ticks);
 
 	if (!atomic_test_and_set_bit(z_power_states_forced, id)) {
-		z_power_states[id] = pm_policy_next_state(id, ticks);
+		const struct pm_state_info *info;
+
+		info = pm_policy_next_state(id, ticks);
+		if (info == NULL) {
+			struct pm_state_info active = {.state = PM_STATE_ACTIVE};
+
+			z_power_states[id] = active;
+		} else {
+			z_power_states[id] = *info;
+		}
```

Comment:
The current power state is `PM_STATE_ACTIVE`, you don't need the assignment when it policy returns `NULL`

Reply:
This is keeping current behavior: as of today, the state returned by `pm_policy_next_state` is always copied to `z_power_states[id]`, even when it is `PM_STATE_ACTIVE`. I'm planning to improve `pm.c` after this.

Reply:
what I saying is that when this function is called, `z_power_states[id]` is `PM_STATE_ACTIVE`. So, when the policy returns `NULL` you don't need to this assigment.
The previous assignment is done because it was cheaper the assignment than a branch + assignement. But now, how it is necessary the branch to check if the policy returned `NULL`, you no longer it.

Reply:
that makes sense - https://github.com/zephyrproject-rtos/zephyr/pull/41703

Fixed the initialization, the cpu state should always be `ACTIVE` when `pm_system_suspend` is called and it can be simplified to:

````
if (info != NULL) {
    z_power_states[id] = *info;
}
````

Reply:
adjusted, https://github.com/zephyrproject-rtos/zephyr/pull/41703 will make sure that this is correct.